internal/cmdsearch/cmdsearch.go

```diff
@@ -86,21 +90,34 @@ func (r *SearchRunner) preRunE(c *cobra.Command, args []string) error {
 		c.Flag("by-value-regex").Changed {
 		return errors.Errorf(`only one of ["by-value", "by-value-regex"] can be provided`)
 	}
+	r.Writer = c.OutOrStdout()
 	return nil
 }
 
 func (r *SearchRunner) runE(c *cobra.Command, args []string) error {
 	e := runner.ExecuteCmdOnPkgs{
-		Writer:             c.OutOrStdout(),
+		Writer:             ioutil.Discard, // dummy writer, runner need not print any info
 		RecurseSubPackages: r.RecurseSubPackages,
 		CmdRunner:          r,
 		RootPkgPath:        args[0],
 		NeedOpenAPI:        true,
+		SkipPkgPathPrint:   true,
 	}
-	return e.Execute()
+	err := e.Execute()
+	if err != nil {
+		return err
+	}
+	var action string
+	if r.PutPattern != "" || r.PutLiteral != "" {
+		action = "Mutated"
+	} else {
+		action = "Matched"
+	}
+	fmt.Fprintf(r.Writer, "%s %d field(s)\n", action, r.MatchCount)
```

Comment:
nit: Ideally I think we should strive for a clearer separation between running the command and printing. So maybe a `print` function that is responsible for printing and eventually could be extended to support different print strategies.